src/Clash/Cores/Ethernet/Examples/TxStack.hs

```diff
@@ -1,49 +1,136 @@
-{-# language FlexibleContexts #-}
-
 {-|
-Module      : Clash.Cores.Ethernet.Examples.TxStack
-Description : Provides the entire transmit stack as a circuit.
+  Module      : Clash.Cores.Ethernet.Examples.TxStack
+  Copyright   : (C) 2024, Tim Wallet <tw.personal@tutanota.com>
+  Description : Provides a standard Ethernet MAC transmit stack.
+  License     : BSD2 (see the file LICENSE)
+  Maintainer  : QBayLogic B.V. <devops@qbaylogic.com>
+
+A fully modular MAC transmit stack which allows the transmission of data over Ethernet II.
+Supports any data width bigger than zero.
+
+Example usage:
+
+>>> import Clash.Prelude
+>>> import Protocols
+>>> import Protocols.Extra.PacketStream
+>>> import Clash.Cores.Ethernet.Mac.EthernetTypes
+
+The Ethernet TX PHY is completely interchangeable with this stack. In the example below,
+we use a dummy. You have to replace this dummy variable with an Ethernet TX PHY circuit
+for your specific hardware (e.g. RGMII, MII or SGMII) that is adapted to the
+`PacketStream` protocol, i.e. with type:
+
+>>> :{
+dummyTxPhy :: Circuit (PacketStream domEthTx 1 ()) (PacketStream domEthTx 1 ())
+dummyTxPhy = undefined
+:}
+
+`txStack` is the most common ethernet MAC TX stack that will be sufficient for
+most people. That is, it inserts an interpacket gap of 12 bytes, pads the payload
+to 46 bytes and assumes that you process the bytes in a different clock domain than the
+ethernet TX domain. All you have to do is specify the data width (in this example 4),
+the clock domains, and the TX PHY you want to use.
+
+>>> :{
+myTxStack :: Circuit (PacketStream dom 4 EthernetHeader) (PacketStream domEthTx 1 ())
+myTxStack = txStack @4 |> dummyTxPhy
+}
+
+Need a TX stack that does it a little different? In this case, you can easily create a
+custom stack by importing the individual components and connecting them via the `|>`
+operator, creating one big `Circuit`. For example:
+
+>>> import Protocols.Extra.PacketStream.Packetizers
+>>> import Clash.Cores.Ethernet.Mac.FrameCheckSequence
+>>> import Clash.Cores.Ethernet.Mac.InterpacketGapInserter
+>>> import Clash.Cores.Ethernet.Mac.PaddingInserter
+>>> import Clash.Cores.Ethernet.Mac.Preamble
+
+This custom TX stack processes bytes in the ethernet TX domain. In this case, we can omit
+`asyncFifoC` and `downConverterC`. We also use a bigger interpacket gap than usual, i.e. 16 bytes.
+
+>>> :{
+myCustomTxStack :: Circuit (PacketStream domEthTx 1 EthernetHeader) (PacketStream domEthTx 1 ())
+myCustomTxStack =
+  macPacketizerC
+  |> paddingInserterC d60
+  |> fcsInserterC
+  |> preambleInserterC
+  |> (exposeClockResetEnable interpacketGapInserterC ethTxClk ethTxRst ethTxEn) d16
```

Comment:
I don't think the `exposeClockResetEnable` is required here? In fact I'm curious how it can work. I would have expected a `HiddenClockResetEnable domEthTx` here for `myCustomTxStack`